src/main/java/com/google/api/gax/core/AbstractApiFuture.java

```diff
@@ -77,35 +76,37 @@ public boolean setException(Throwable throwable) {
     return impl.setException(throwable);
   }
 
-  public void addCallback(final RpcFutureCallback<? super V> callback) {
-    Futures.addCallback(
-        impl,
-        new FutureCallback<V>() {
-          @Override
-          public void onFailure(Throwable t) {
-            callback.onFailure(t);
-          }
-
-          @Override
-          public void onSuccess(V v) {
-            callback.onSuccess(v);
-          }
-        });
+  protected void interruptTask() {}
+
+  /* package-private for internal library use, to reduce call layering */
+  ListenableFuture<V> getInternalListenableFuture() {
+    return impl;
   }
 
-  public <X extends Throwable> RpcFuture catching(
-      Class<X> exceptionType, final Function<? super X, ? extends V> callback) {
-    SettableRpcFuture<V> future = new SettableRpcFuture<>();
-    future.impl.setFuture(
-        Futures.catching(
-            impl,
-            exceptionType,
-            new com.google.common.base.Function<X, V>() {
-              @Override
-              public V apply(X input) {
-                return callback.apply(input);
-              }
-            }));
-    return future;
+  private class InternalSettableFuture extends AbstractFuture<V> {
+    @Override
+    public final V get() throws InterruptedException, ExecutionException {
```

Comment:
`final` is definitely not necessary. Having the `get` methods here is a carryover from what is in Guava, but it might not be necessary here.